common/djangoapps/student/models.py

```diff
@@ -2177,12 +2177,17 @@ def get_manual_enrollment(cls, enrollment):
         return manual_enrollment
 
     @classmethod
-    def retire_manual_enrollments(cls, enrollments, retired_email):
+    def retire_manual_enrollments(cls, user, retired_email):
         """
-        Removes PII (enrolled_email and reason) from any rows corresponding to
-        the enrollment passed in. Bubbles up any exceptions.
+        Removes PII (enrolled_email and reason) associated with the User passed in. Bubbles up any exceptions.
         """
-        return cls.objects.filter(enrollment__in=enrollments).update(reason="", enrolled_email=retired_email)
+        # This bit of ugliness is to fix a perfmance issue with Django using a slow
+        # sub-select that caused the original query to take several seconds (PLAT-2371).
+        # It is possible that this could also be bad if a user has thousands of manual
+        # enrollments, but currently that number tends to be very low.
+        manual_enrollment_ids = list(cls.objects.filter(enrollment__user=user).values_list('id', flat=True))
```

Comment:
Would it be bad in that the thousands of enrollment ids would eat up python memory?

Reply:
Yes, that's what I meant in the comment. However these are only the manual enrollments not all of the user's enrollments, so I would be very surprised if we had any over double-digits. It's also limited to just the ids coming back, not the whole model object, so even 10k manual enrollments should still only be something like 300k.